### PR TITLE
Add children from school

### PR DESCRIPTION
**Pull request by kirtijadhav**
see issue: #944 

### Visible/Frontend Changes
- [ ] 
- [ ] 

### Architectural/Backend Changes
- [x] add child new button
- [ ]

### CODE REVIEW

File: src/app/child-dev-project/children/model/childSchoolRelation.ts [hunk 1]
```diff
@@ -3,13 +3,20 @@ import { DatabaseEntity } from "../../../core/entity/database-entity.decorator";
...
+import { Child } from "./child";
 
 /**
  * Record of a school year that a Child attended a certain class in a School.
  */
 @DatabaseEntity("ChildSchoolRelation")
 export class ChildSchoolRelation extends Entity {
-  @DatabaseField() childId: string;
+  @DatabaseField({
+    label: $localize`:Label for the child of a relation:Child`,
```
Comment (TheSlimvReal) [inline src/app/child-dev-project/children/model/childSchoolRelation.ts:14]: A german translation has to be provided for this

File: src/app/child-dev-project/schools/children-overview/children-overview.component.html [hunk 2]
```diff
@@ -0,0 +1,19 @@
+<button
+  mat-stroked-button
+  color="accent"
+  (click)="addChildClick()"
+>
+  <mat-icon style="font-size: 1.49em;"
+    class="button-icon"
+    aria-label="add child"
+    fontIcon="fa-plus-circle"
+  ></mat-icon>
```
Comment (TheSlimvReal) [inline src/app/child-dev-project/schools/children-overview/children-overview.component.html:10]: The styling of the '+' button is a little off

add_child_button

File: src/app/child-dev-project/schools/children-overview/children-overview.component.ts [hunk 3]
```diff
@@ -35,19 +35,41 @@ export class ChildrenOverviewComponent implements OnInitDynamicComponent {
...
+      this.entity.getId()
     );
   }
 
   routeToChild(child: Child) {
     this.router.navigate([`/${child.getType().toLowerCase()}`, child.getId()]);
   }
+
+  addChildClick() {
+    console.log('addChildClick')
```
Comment (TheSlimvReal) [inline src/app/child-dev-project/schools/children-overview/children-overview.component.ts:57]: remove this line

File: src/app/child-dev-project/children/model/childSchoolRelation.ts (hunk 1 above)
Comment (TheSlimvReal) [reply src/app/child-dev-project/children/model/childSchoolRelation.ts:14]: See our guide for [building localizable components](https://aam-digital.github.io/ndb-core/documentation/additional-documentation/how-to-guides/build-localizable-components.html)

**Review by TheSlimvReal — CHANGES_REQUESTED** [review]

**Review by TheSlimvReal — CHANGES_REQUESTED** [review]
Looks very good but there are some problems:

- [ ] Since autocomplete was added, the children are not added to the list any more
- [ ] The label is very specific to the children case, we should find a way to make if configurable

File: src/app/core/entity-components/entity-utils/dynamic-form-components/edit-single-entity/edit-single-entity.component.html [hunk 4]
```diff
@@ -1,11 +1,23 @@
...
+    [placeholder]="placeholder"
+    [formControl]="formControl"
+    [matAutocomplete]="autoSuggestions"
+  />
+  <mat-autocomplete
+    #autoSuggestions="matAutocomplete"
+  >
+    <!-- Optional header -->
+    <ng-content select="mat-option"></ng-content>
+    <mat-option *ngFor="let getId of filteredChildren | async" [value]="getId">
```
Comment (TheSlimvReal) [inline src/app/core/entity-components/entity-utils/dynamic-form-components/edit-single-entity/edit-single-entity.component.html:16]:
```suggestion
    <mat-option *ngFor="let child of filteredChildren | async" [value]="child.getId()">
```

File: src/app/core/entity-components/entity-utils/dynamic-form-components/edit-single-entity/edit-single-entity.component.html [hunk 5]
```diff
@@ -1,11 +1,23 @@
...
       <app-display-entity
-        [entityToDisplay]="entity"
+        [entityToDisplay]="getId"
```
Comment (TheSlimvReal) [inline src/app/core/entity-components/entity-utils/dynamic-form-components/edit-single-entity/edit-single-entity.component.html:18]:
```suggestion
        [entityToDisplay]="child"
```

**Review by TheSlimvReal — CHANGES_REQUESTED** [review]
I guess there was a small misunderstanding. The important part is that just the ID and not the whole entity is assigned as value.

**Comment by TheSlimvReal** [comment]
The functionality works well now.
The problem is however, that the ID of the child is shown in the input field rather then the name. This should be changed.

**Review by TheSlimvReal — APPROVED** [review]
Good job, works well now

**Comment by sleidig** [comment]
closes #758

**Pull request merged**